activestorage/app/controllers/concerns/active_storage/streaming.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+module ActiveStorage::Streaming
+  private
+
+  # Stream the blob from storage directly to the response. The disposition can be controlled by setting +disposition+.
+  # The content type and filename is set directly from the +blob+.
+  def stream_from_storage(blob, disposition: "inline")
+    response.headers["Content-Type"] = blob.content_type_for_serving
+    response.headers["Content-Disposition"] = ActionDispatch::Http::ContentDisposition.format \
+      disposition: blob.forced_disposition_for_serving || disposition, filename: blob.filename.sanitized
+
+    blob.download do |chunk|
+      response.stream.write chunk
+    end
```

Comment:
Now that we're on 2.7 I think we could try the shorthand method#to_proc reference:

```suggestion
    blob.download(&response.stream.:write)
```

I feel sorta ambivalent, but there's something interesting to it.

Reply:
imo too clever